mm2src/coins/lightning/ln_platform.rs

```diff
@@ -0,0 +1,688 @@
+                            let history = client
+                                .scripthash_get_history(&script_hash)
+                                .compat()
+                                .await
+                                .unwrap_or_default();
+                            for item in history {
+                                if item.tx_hash == rpc_txid {
+                                    // If a new block mined the transaction while running process_txs_confirmations it will be confirmed later in ln_best_block_update_loop
+                                    if item.height > 0 && item.height <= current_height as i64 {
+                                        let height: u64 = match item.height.try_into() {
+                                            Ok(h) => h,
+                                            Err(e) => {
+                                                log::error!("Block height convertion to u64 error: {}", e.to_string());
+                                                continue;
+                                            },
+                                        };
+                                        let header = match client.blockchain_block_header(height).compat().await {
+                                            Ok(block_header) => match deserialize(&block_header) {
+                                                Ok(h) => h,
+                                                Err(e) => {
+                                                    log::error!(
+                                                        "Block header deserialization error: {}",
+                                                        e.to_string()
+                                                    );
+                                                    continue;
+                                                },
+                                            },
+                                            Err(_) => continue,
+                                        };
+                                        let index = match client
+                                            .blockchain_transaction_get_merkle(rpc_txid, height)
+                                            .compat()
+                                            .await
+                                        {
+                                            Ok(merkle_branch) => merkle_branch.pos,
+                                            Err(e) => {
+                                                log::error!(
+                                                    "Error getting transaction position in the block: {}",
+                                                    e.to_string()
+                                                );
+                                                continue;
+                                            },
+                                        };
```

Comment:
This function also can be simplified by using a macro like this:
https://github.com/KomodoPlatform/atomicDEX-API/blob/dbacb816c3ece306628ca7517ef57c29036e0b1e/mm2src/coins/utxo/rpc_clients.rs#L1748-L1759